lib/iris/tests/test_constraints.py

```diff
@@ -295,50 +296,241 @@ def load_match(self, files, constraints):
         return cubes
 
 
+@tests.skip_data
+class TestCubeExtract__names(TestMixin, tests.IrisTest):
+    def setUp(self):
+        TestMixin.setUp(self)
+        self.cube = iris.load_cube(self.theta_path)
+        # Expected names...
+        self.standard_name = "air_potential_temperature"
+        self.long_name = "air potential temperature"
+        self.var_name = "apt"
+        self.stash = "m01s00i004"
+        # Configure missing names...
+        self.cube.long_name = self.long_name
+        self.cube.var_name = self.var_name
+
+    def test_standard_name(self):
+        constraint = iris.Constraint(self.standard_name)
+        result = self.cube.extract(constraint)
+        self.assertIsNotNone(result)
+        self.assertEqual(result.standard_name, self.standard_name)
+
+    def test_long_name(self):
+        constraint = iris.Constraint(self.long_name)
+        result = self.cube.extract(constraint)
+        self.assertIsNotNone(result)
+        self.assertEqual(result.long_name, self.long_name)
+
+    def test_var_name(self):
+        constraint = iris.Constraint(self.var_name)
+        result = self.cube.extract(constraint)
+        self.assertIsNotNone(result)
+        self.assertEqual(result.var_name, self.var_name)
+
+    def test_stash(self):
+        constraint = iris.Constraint(self.stash)
+        result = self.cube.extract(constraint)
+        self.assertIsNotNone(result)
+        self.assertEqual(str(result.attributes["STASH"]), self.stash)
+
+    def test_unknown(self):
+        # Clear the cube metadata.
```

Comment:
I think it would be worth showing first, that Constraint("unknown") would *not* match the cube before all its names were removed.

Reply:
Still not doing this AFAICT
? did this get lost in a rewrite ?

Reply:
:+1: now testing this, due to other changes